options.go

```diff
@@ -216,12 +216,17 @@ func WithPartitionChannelSize(size int) ProcessorOption {
 
 // WithLogger sets the logger the processor should use. By default, processors
 // use the standard library logger.
-func WithLogger(l Logger) ProcessorOption {
+// To enable debug logging, pass true as an optional parameter
+func WithLogger(l Logger, debug ...bool) ProcessorOption {
 	return func(o *poptions, gg *GroupGraph) {
 		if prefixLogger, ok := l.(logger); ok {
 			o.log = prefixLogger
 		} else {
-			o.log = wrapLogger(l)
+			debugLog := defaultLogger.debug
+			if len(debug) > 0 {
+				debugLog = debug[0]
+			}
+			o.log = wrapLogger(l, debugLog)
```

Comment:
I think it's a litte bit confusing. 
As an user of the function I would be confused why I should add a list of bools.
Is this a common pattern to design an optional parameter?

I get that we have to make the code backwards compatible, but wouldn't be a new options function cleaner?
e.g. `WithDebugLogger()`

Reply:
yeah you're right, it's too confusing and the pattern is not really established. Having a `WithDebugLogger` option is also not optimal because one would have to change the processor definition (which might be hidden in helper functions) just to enable logging. So I removed the extra option, and the `WithLogger` option just inherits the goka-global debug setting.